src/daft-core/src/utils/stats.rs

```diff
@@ -0,0 +1,82 @@
+use common_error::DaftResult;
+
+use crate::{
+    array::{
+        ops::{DaftCountAggable, DaftSumAggable, GroupIndices, VecIndices},
+        prelude::{Float64Array, UInt64Array},
+    },
+    count_mode::CountMode,
+};
+
+#[derive(Clone, Copy, Default, Debug)]
+pub struct Stats {
+    pub sum: f64,
+    pub count: u64,
+    pub mean: Option<f64>,
+}
+
+pub fn calculate_stats(array: &Float64Array) -> DaftResult<Stats> {
+    let sum = array.sum()?.get(0);
+    let count = array.count(CountMode::Valid)?.get(0);
+    let stats = sum
+        .zip(count)
+        .map_or_else(Default::default, |(sum, count)| Stats {
+            sum,
+            count,
+            mean: calculate_mean(sum, count),
+        });
+    Ok(stats)
+}
+
+pub fn grouped_stats<'a>(
+    array: &Float64Array,
+    groups: &'a GroupIndices,
+) -> DaftResult<impl Iterator<Item = (Stats, &'a VecIndices)>> {
+    let grouped_sum = array.grouped_sum(groups)?;
+    let grouped_count = array.grouped_count(groups, CountMode::Valid)?;
+    assert_eq!(grouped_sum.len(), grouped_count.len());
+    assert_eq!(grouped_sum.len(), groups.len());
```

Comment:
should these be debug assertions instead?

Reply:
Sure, can turn them into debug assertions! I just saw them in the previous implementation, so I decided to carry them forwards.

Will turn it into something like:

```rs
#[cfg(debug_assertions)]
assert_eq!(..);
```

Reply:
Btw I think we already have `debug_assert!`?

Reply:
Oh sick! I'll use that instead then.